src/ServiceControl/Monitoring/HeartbeatMonitoring/HeartbeatTypesArrayToInstanceConverter.cs

```diff
@@ -43,7 +43,7 @@ sealed class Converter<TValue>(JsonSerializerOptions options) : JsonConverter<TV
 
                     if (reader.TokenType != JsonTokenType.StartObject)
                     {
-                        throw new JsonException();
+                        throw new JsonException($"Unsupported structure for {type}");
```

Comment:
This change is usually incorrect. The JsonException does give enough context automatically. All the guidance of MS uses just that. This should be undone